Update README and HelloPhi example (#477)
microsoft/onnxruntime-genai: README.md

|Support matrix|Supported now|Under development|On the roadmap|
|-|-|-|-|
|Model architectures|  Gemma <br/> Llama * <br/> Mistral + <br/>Phi <br/>|Whisper|Stable diffusion|
|API| Python <br/>C# <br/>C/C++ |Java|Objective-C|||
|Platform| Linux <br/> Windows  | Android | Mac <br/> iOS |||
|Architecture|x86 <br/> x64 <br/> | Arm64 |||
|Hardware Acceleration|CUDA<br/>DirectML<br/>|QNN <br/> ROCm |OpenVINO

> \* The Llama model architecture also includes CodeLlama models, Yi models and Vicuna models

Reviewer: ```suggestion
\* The Llama model architecture supports similar model families such as CodeLlama, Vicuna, Yi, and more.
```

> \+ The Mistral model architecture also includes Zephyr models

Reviewer: ```suggestion
\+ The Mistral model architecture supports similar model families such as Zephyr.
```
